check_whitespace.c

```diff
@@ -66,7 +67,10 @@ int is_clean(char* str) {
   // 0 if they're equal, and a positive value if the first is
   // greater than the second.
   result = strcmp(str, cleaned);
-
+  //free(cleaned);
+  if (strlen(cleaned) != 0) {
```

Comment:
Alternatively you could use `strcmp`, which _may_ make it clearer what your intent is? (Also that's arguably a matter of opinion.)

```suggestion
  if (strcmp(cleaned, "") != 0) {
```